src/strings/unicode_ops.c

```diff
@@ -1,3 +1,56 @@
+/* Compares two strings, returning -1, 0 or 1 to indicate less than,
+ * equal or greater than based on Unicode Collation Algorithm */
+MVMint64 MVM_unicode_string_compare(MVMThreadContext *tc, MVMString *a, MVMString *b) {
+    MVMStringIndex alen, blen, i, scanlen;
+    // Interation variables
+    MVMGrapheme32 ai;
+    MVMGrapheme32 bi;
+    // Colation order numbers
+    MVMint32 ai_coll_val;
+    MVMint32 bi_coll_val;
+    MVM_string_check_arg(tc, a, "compare");
+    MVM_string_check_arg(tc, b, "compare");
+    /* Simple cases when one or both are zero length. */
+    alen = MVM_string_graphs(tc, a);
+    blen = MVM_string_graphs(tc, b);
+    if (alen == 0)
+        return blen == 0 ? 0 : -1;
+    if (blen == 0)
+        return 1;
+
+    /* Otherwise, need to scan them. */
+    scanlen = alen > blen ? blen : alen;
+    for (i = 0; i < scanlen; i++) {
+        ai = MVM_string_get_grapheme_at_nocheck(tc, a, i);
+        bi = MVM_string_get_grapheme_at_nocheck(tc, b, i);
```

Comment:
We use `MVM_string_get_grapheme_at_nocheck` for the normal string comparing function. Would it be proper for them both to use GraphemeIter instead?